packages/blade/src/components/ProgressBar/ProgressBar.tsx

```diff
@@ -0,0 +1,142 @@
+
+type ProgressBarCommonProps = {
+  accessibilityLabel?: string;
+  contrast?: ColorContrastTypes;
+  intent?: Feedback;
+  label?: string;
+  showPercentage?: boolean;
+  size?: 'small' | 'medium';
+  value?: number;
+};
+
+type ProgressBarVariant = 'progress' | 'meter';
+
+type ProgressBarProgressProps = ProgressBarCommonProps & {
+  variant?: Extract<ProgressBarVariant, 'progress'>;
+  isIndeterminate?: boolean;
+};
+
+type ProgressBarMeterProps = ProgressBarCommonProps & {
+  variant?: Extract<ProgressBarVariant, 'meter'>;
+  isIndeterminate?: undefined;
+};
+
+type ProgressBarProps<T> = T extends {
+  variant: infer Variant;
+}
+  ? Variant extends 'progress'
+    ? ProgressBarProgressProps
+    : Variant extends 'meter'
+    ? ProgressBarMeterProps
+    : T
+  : T;
+
+type ProgressBarUnfilledProps = {
+  backgroundColor: string;
+};
+
+const ProgressBarUnfilled = styled(Box)<ProgressBarUnfilledProps>(({ backgroundColor }) => ({
+  backgroundColor,
+}));
+
+const getProgress = (value: number): number => {
+  return Math.floor(Math.min(100, Math.max(0, value)));
+};
+
+const progressBarHeight: Record<NonNullable<ProgressBarCommonProps['size']>, 2 | 4> = {
+  small: 2,
+  medium: 4,
+};
+
+const ProgressBar = <T extends { variant: ProgressBarVariant }>({
+  accessibilityLabel,
+  contrast = 'low',
+  intent,
+  isIndeterminate,
+  label,
+  showPercentage = false,
+  size = 'small',
+  value = 0,
+  variant = 'progress',
+}: ProgressBarProps<T>): ReactElement => {
+  console.log(isIndeterminate); // will be in follow up PR
+
+  if (variant === 'meter' && isIndeterminate) {
+    console.warn(`[Blade: ProgressBar]: Cannot set 'isIndeterminate' when 'variant' is 'meter'.`);
+  }
+
+  const { theme } = useTheme();
+  const unfilledBackgroundColor = theme.colors.brand.gray.a100[`${contrast}Contrast`];
+  const filledBackgroundColor = intent
+    ? theme.colors.feedback.background[intent].highContrast
+    : theme.colors.brand.primary[500];
+  const hasLabel = label && label.trim()?.length > 0;
+  const isMeter = variant === 'meter';
+  const progressValue = getProgress(value);
+  const id = useId(variant);
+
+  return (
+    <Box>
+      <Box
+        display="flex"
+        flexDirection="row"
+        justifyContent={hasLabel ? 'space-between' : 'flex-end'}
+      >
+        {hasLabel ? (
+          <FormLabel as="label" htmlFor={id} contrast={contrast}>
+            {label}
+          </FormLabel>
+        ) : null}
+        {showPercentage ? (
+          <Box marginBottom="spacing.2">
+            <Text
+              type="subdued"
+              variant="body"
+              contrast={contrast}
+              size="small"
+            >{`${progressValue}%`}</Text>
+          </Box>
+        ) : null}
+      </Box>
+      <Box
+        id={id}
+        {...makeAccessible({
+          role: variant === 'meter' ? 'meter' : 'progressbar',
+          label: accessibilityLabel ?? label,
+          busy: !isMeter,
```

Comment:
aria-busy should not be applied to the progressbar, aria-busy should be applied to the region which is being loaded. 

For example if a infinite scroll feed is loading aria-busy will be added to the whole infinite scrolling div.

Reply:
Fair. The mdn docs [say the same](https://developer.mozilla.org/en-US/docs/Web/Accessibility/ARIA/Roles/progressbar_role#:~:text=set%20the%20aria%2Dbusy%20attribute%20to%20true%20on%20the%20region%20until%20it%20is%20finished%20loading.).  Although I looked at a few other design systems around and some of them had aria-busy set to true. Better to not add it here and let the consumers add it in the region then.